### PR TITLE
Make AdminClient a module

### PR DESCRIPTION
**Pull request by gurghet**
I just mechanically copied the work done on consumer and producer.

### CODE REVIEW

File: src/main/scala/zio/kafka/admin/AdminClient.scala [hunk 1]
```diff
@@ -247,47 +145,271 @@ case class AdminClient(private val adminClient: JAdminClient, private val blocki
...
+}
+
+object AdminClient {
+
+  /**
+   * Thin wrapper around apache java AdminClient. See java api for descriptions
+   * @param adminClient
+   */
+  private final case class LiveAdminClient(
+    private[admin] val adminClient: JAdminClient,
```
Comment (gurghet) [inline src/main/scala/zio/kafka/admin/AdminClient.scala:167]: Any nicer way to do this?

File: src/main/scala/zio/kafka/admin/AdminClient.scala [hunk 2]
```diff
@@ -622,10 +744,10 @@ object AdminClient {
       KafkaConfig(jConfig.entries().asScala.map(e => e.name() -> e).toMap)
   }
 
-  def make(settings: AdminClientSettings) =
+  def make(settings: AdminClientSettings): ZManaged[Has[Blocking.Service], Throwable, AdminClient] =
```
Comment (gurghet) [inline src/main/scala/zio/kafka/admin/AdminClient.scala:747]: Without explicit type it says I'm exporting `LiveAdminClient`, I don't know if is supposed to.

Comment (iravid) [reply src/main/scala/zio/kafka/admin/AdminClient.scala:747]: Not having the type annotation there is an oversight - thank you for adding it!

File: src/main/scala/zio/kafka/admin/AdminClient.scala (hunk 1 above)
Comment (iravid) [reply src/main/scala/zio/kafka/admin/AdminClient.scala:167]: This looks pretty good to me. What are you referring to specifically?

**Review by iravid — COMMENTED** [review]
Awesome work. Let's also add a `def live: ZLayer[Has[Blocking.Service], Throwable, Has[AdminClient]]` on the `AdminClient` object?

**Comment by gurghet** [comment]
I added `def live: ZLayer[Has[Blocking.Service] with Has[AdminClientSettings], Throwable, Has[AdminClient]]` instead of passing the `AdminClientSettings` as a parameter. I don't completely understand why this was done and did it for sake of uniformity.

File: src/main/scala/zio/kafka/admin/AdminClient.scala (hunk 1 above)
Comment (gurghet) [reply src/main/scala/zio/kafka/admin/AdminClient.scala:167]: thinking about it, it actually looks ok, I guess I was bothered by the asymmetry of the two params scope.

File: src/main/scala/zio/kafka/admin/AdminClient.scala [hunk 3]
```diff
@@ -247,47 +145,277 @@ case class AdminClient(private val adminClient: JAdminClient, private val blocki
...
   }
-}
 
-object AdminClient {
+  val live: ZLayer[Has[Blocking.Service] with Has[AdminClientSettings], Throwable, Has[AdminClient]] =
+    (for {
+      settings <- ZManaged.service[AdminClientSettings]
+      admin    <- make(settings)
+    } yield admin).toLayer
+
```
Comment (gurghet) [inline src/main/scala/zio/kafka/admin/AdminClient.scala:418]: `live` helper function, I guess one could always use the make function if he wants to pass the `settings` by parameter.

**Comment by iravid** [comment]
Anything else you'd like to add here @gurghet? Or can we merge and release this?

**Comment by gurghet** [comment]
No, it's fine :shipit:

**Review by iravid — APPROVED** [review]

**Pull request merged**